### ISSUE TITLE
Impossible to attach arbitrary stacktrace to log message when using `:logger` handler

### ISSUE THREAD

**Issue by martosaur** [getsentry/sentry-elixir#710]
`LoggerBackend` has this nice, even if unintentional, feature that it reads from `meta[:sentry]` thus allowing users to enrich log messages with arbitrary stacktraces:

```elixir
Logger.error("Log with stacktrace", sentry: %{stacktrace: Process.info(self(), :current_stacktrace) |> elem(1)})
```

![CleanShot 2024-04-03 at 11 36 10](https://github.com/getsentry/sentry-elixir/assets/5783386/0bfa416e-3dec-4b1a-bbe7-f63b87ee6c37)

`LoggerHandler`  on the other hand very explicitly ignores `meta[:sentry]` which AFAIK leaves no hatch to pass a stacktrace.
https://github.com/getsentry/sentry-elixir/blob/3e820f789cb67eac9a1a8dc5939e82f223259c90/lib/sentry/logger_handler.ex#L148-L153

It would be nice to provide some avenue for overriding Sentry context on per-message basis. WDYT?

**Comment by whatyouhide** [comment]
Hi, can you try the `al/logger-backend-meta` branch from GitHub in your project? Should fix that.

```elixir
  {:sentry, github: "getsentry/sentry-elixir", branch: "al/logger-backend-meta"}
```

**Comment by martosaur** [comment]
@whatyouhide Thanks for looking into this! I'm on vacation now away from my laptop, but I'll test it as soon as I'm back! Hope it's not too much of a delay ❤️

**Comment by whatyouhide** [comment]
@martosaur I believe the fix in #712 is correct, so you can try it out on the `main` branch now. Feel free to comment here if anything looks off. Thanks!

**Comment by martosaur** [comment]
Yep it does work as expected, thank you!
